### PR TITLE
MC-40777: Cannot save product in store view scope without Magento_Catalog::edit_product_design

### LINKED ISSUE #31973: Cannot save product in store view scope without Magento_Catalog::edit_product_design
**Preconditions**

Magento 2.4-develop;
Magento Open Source 2.3.5-p2;

**Steps to reproduce**
1. Create a simple product;
1. Use an administrator account, set the "Schedule Design Update" attribute under the store view scope (***Default Store View*** for ex.)  for any product, and then save changes;
image
1. Go to Admin -> System -> Permissions -> User Roles and create a new role with the entire `Catalog` tree selected, except for `Catalog > Inventory > Products > Edit Product Design`;
Screenshot from 2021-02-05 10-47-15
1. Create a user assigned to this new role and log into the backend with it;
1. Edit the same product, change the product name and save changes under the store view scope (***Default Store View***);

**Expected result**

(/) The changes can be saved successfully.

**Actual result**

(x) Raise an error and failed to save.
Error message: Not allowed to edit the product's design attributes
image

In the ***exception.log*** file:
Screenshot from 2021-02-05 11-10-44

### PR DESCRIPTION
**Pull request by engcom-Echo**
### Fixed Issues (if relevant)

1. Fixes magento/magento2#31973

### Manual testing scenarios (*)

1. See magento/magento2#31973

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run all tests

File: app/code/Magento/Catalog/Model/Product/Authorization.php [hunk 1]
```diff
@@ -32,14 +34,24 @@ class Authorization
...
      * @param AuthorizationInterface $authorization
      * @param ProductFactory $factory
+     * @param RequestInterface|null $request
      */
-    public function __construct(AuthorizationInterface $authorization, ProductFactory $factory)
-    {
+    public function __construct(
+        AuthorizationInterface $authorization,
+        ProductFactory $factory,
+        ?RequestInterface $request = null
```
Comment (zakdma) [inline app/code/Magento/Catalog/Model/Product/Authorization.php]: Dont think you should use Request inside model.
Please change implementation to not involve request to model.

**Review by zakdma — CHANGES_REQUESTED** [review]

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run all tests

File: app/code/Magento/Catalog/Model/Product/Authorization.php [hunk 2]
```diff
@@ -129,6 +129,9 @@ private function hasProductChanged(ProductModel $product, ?array $oldProduct = n
                 //No new value
                 continue;
             }
+            if ($attribute->getBackendType() == 'datetime' && empty($newValue)) {
```
Comment (zakdma) [inline app/code/Magento/Catalog/Model/Product/Authorization.php]: Please change second condition to $newValue === null

File: dev/tests/integration/testsuite/Magento/Catalog/_files/product_simple_with_design_attributes.php [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    ->setMetaDescription('meta description')
+    ->setVisibility(Visibility::VISIBILITY_BOTH)
+    ->setStatus(Status::STATUS_ENABLED)
+    ->setStockData(['use_config_manage_stock' => 0])
+    ->setCanSaveCustomOptions(true)
+    ->setCustomDesignFrom('2020-01-02')
+    ->setCustomDesignTo('2020-01-03')
+    ->setHasOptions(true);
+/** @var ProductRepositoryInterface $productRepositoryFactory */
+$productRepository = $objectManager->create(ProductRepositoryInterface::class);
```
Comment (zakdma) [inline dev/tests/integration/testsuite/Magento/Catalog/_files/product_simple_with_design_attributes.php]: Please use get()

**Review by zakdma — CHANGES_REQUESTED** [review]

**Comment by engcom-Echo** [comment]
@magento run all tests

**Review by zakdma — APPROVED** [review]

**Comment by dmitriyprime** [comment]
@magento I am working on this

**Comment by dmitriyprime** [comment]
:heavy_minus_sign:  QA failed
If administrator with restricted role changes on Schedule Design Update tab not only Schedule Update From field but for example New Theme or New Layout fields or both he/she cannot save product after changing some product attributes (for example product name) on specific store view.

**Comment by engcom-Echo** [comment]
@magento run all tests

**Review by zakdma — APPROVED** [review]

**Comment by engcom-Oscar** [comment]
:heavy_check_mark: QA passed

Was checked case from Manual testing scenarios 

:x: **Before:**

Screenshot from 2021-02-18 15-20-31

:heavy_check_mark: **After:**

Screenshot from 2021-02-18 15-44-40

**Comment by zakdma** [comment]
@magento import code to https://github.com/magento-tsg/magento2ce

**Comment by magento-engcom-team** [comment]
@zakdma the branch with code successfully imported into`magento-tsg/magento2ce` repository. Branch name: `imported-magento-magento2-31998`.

**Pull request merged**